### PR TITLE
Add user attributes to endpoint profile user

### PR DESCRIPTION
**Pull request by raphkim**
*Issue #, if available:*
[Internal ticket](https://t.corp.amazon.com/mobilesdk-4751)

*Description of changes:*
Add `userAttributes` field to `EndpointProfileUser` to better represent `EndpointUser` model. This field will now update alongside endpoint profile.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: aws-android-sdk-pinpoint-test/src/androidTest/java/com/amazonaws/mobileconnectors/pinpoint/analytics/EndpointProfileIntegrationTest.java [hunk 1]
```diff
@@ -106,16 +107,21 @@ public void testEndpointProfileUpdate() {
         EndpointProfile endpointProfile = targetingClient.currentEndpoint();
         assertNotNull(endpointProfile);
         assertNull(endpointProfile.getUser().getUserId());
+        assertNull(endpointProfile.getUser().getUserAttributes());
 
         EndpointProfileUser user = new EndpointProfileUser();
         user.setUserId(credentialsProvider.getIdentityId());
+        user.setUserAttributes(Collections.singletonMap("user-key", Arrays.asList("user-value")));
```
Comment (jamesonwilliams) [inline aws-android-sdk-pinpoint-test/src/androidTest/java/com/amazonaws/mobileconnectors/pinpoint/analytics/EndpointProfileIntegrationTest.java]: `Collections.singletonList("user-value")`

File: aws-android-sdk-pinpoint-test/src/androidTest/java/com/amazonaws/mobileconnectors/pinpoint/analytics/EndpointProfileIntegrationTest.java [hunk 2]
```diff
@@ -106,16 +107,21 @@ public void testEndpointProfileUpdate() {
...
         endpointProfile.setUser(user);
 
         targetingClient.updateEndpointProfile();
         endpointProfile = targetingClient.currentEndpoint();
         assertNotNull(endpointProfile);
         assertEquals(credentialsProvider.getIdentityId(),
                 endpointProfile.getUser().getUserId());
+        assertNotNull(endpointProfile.getUser().getUserAttributes());
+        assertEquals(Collections.singletonMap("user-key", Arrays.asList("user-value")),
```
Comment (jamesonwilliams) [inline aws-android-sdk-pinpoint-test/src/androidTest/java/com/amazonaws/mobileconnectors/pinpoint/analytics/EndpointProfileIntegrationTest.java]: Same here, can use `Collections.singletonList("user-value")`

File: aws-android-sdk-pinpoint-test/src/androidTest/java/com/amazonaws/mobileconnectors/pinpoint/analytics/EndpointProfileIntegrationTest.java (hunk 1 above)
Comment (raphkim) [reply aws-android-sdk-pinpoint-test/src/androidTest/java/com/amazonaws/mobileconnectors/pinpoint/analytics/EndpointProfileIntegrationTest.java]: fixed

File: aws-android-sdk-pinpoint-test/src/androidTest/java/com/amazonaws/mobileconnectors/pinpoint/analytics/EndpointProfileIntegrationTest.java (hunk 2 above)
Comment (raphkim) [reply aws-android-sdk-pinpoint-test/src/androidTest/java/com/amazonaws/mobileconnectors/pinpoint/analytics/EndpointProfileIntegrationTest.java]: fixed

**Comment by raphkim** [comment]
Added null-check for user attributes when converting to JSON object

**Pull request merged**